cherrypy/lib/static.py

```diff
@@ -165,7 +167,7 @@ def _serve_fileobj(fileobj, content_type, content_length, debug=False):
     request = cherrypy.serving.request
     if request.protocol >= (1, 1):
         response.headers['Accept-Ranges'] = 'bytes'
-        r = httputil.get_ranges(request.headers.get('Range'), content_length)
+        r = httputil.passes_if_range_check(request) and httputil.get_ranges(request.headers.get('Range'), content_length)
```

Comment:
I don't think it's a good idea to mix different types of data in one variable. Please have separate vars for that.

Reply:
I believe in this case it makes sense, as if `passes_if_range_check` fails, we want the value to be falsy, but if it's true, we want to keep the value of whatever `get_ranges` returns.

I can definitely change it if you'd like!

Reply:
It's an example of code which looks clever at first but is leaking abstractions and is hard to understand/maintain.

Reply:
Sure thing, I'll go ahead and split it off into two different variables. I have a bunch of changes coming in. I accidentally pushed before I was ready trying to get the test suite to run on my new machine.

Reply:
It's actually supported in `caching` tool: https://github.com/cherrypy/cherrypy/search?q=expires&unscoped_q=expires&type=Code

And it just occurred to me that maybe this check should be done there, not in this module.

There's also a tool called `expires` + there's tests for etag as well.

Could you please explore this?